distributed/scheduler.py

```diff
@@ -7736,7 +7737,10 @@ def check_idle(self) -> int | None:
             return self.idle_since
```

Comment:
This will return when the scheduler is idle since, but doesn't take into account Jupyter's idleness.

Reply:
Jupyter's `last_activity` time shouldn't be in the future, so it must be <= `time()`. In other words, `idle_since` is storing `max(dask_last_activity, jupyter_last_activity)`.